src/Controller/CRUDController.php

```diff
@@ -39,18 +39,17 @@
 use Symfony\Component\HttpFoundation\Request;
 use Symfony\Component\HttpFoundation\Response;
 use Symfony\Component\HttpKernel\Exception\HttpException;
-use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
 use Symfony\Component\PropertyAccess\PropertyAccess;
 use Symfony\Component\PropertyAccess\PropertyPath;
-use Symfony\Component\Security\Core\Exception\AccessDeniedException;
 use Symfony\Component\Security\Csrf\CsrfToken;
 
 /**
  * @author Thomas Rabaix <thomas.rabaix@sonata-project.org>
  *
  * @phpstan-template T of object
+ * @phpstan-implements CRUDControllerInterface<T>
  */
-class CRUDController implements ContainerAwareInterface
+class CRUDController implements ContainerAwareInterface, CRUDControllerInterface
```

Comment:
An interface for a controller sounds like over-engineering

Reply:
👍 I did it in a separated commit because I wasn't sure, it's removed.